src/components/UncontrolledForm.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useRef, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from './store';
+import { addFormData } from './store/formSlice';
+
+interface UncontrolledFormProps {
+  onSuccess: () => void;
+}
+
+const UncontrolledForm: React.FC<UncontrolledFormProps> = ({ onSuccess }) => {
+  const dispatch = useDispatch();
+  const countries = useSelector((state: RootState) => state.countries);
+
+  const nameRef = useRef<HTMLInputElement>(null);
+  const ageRef = useRef<HTMLInputElement>(null);
+  const emailRef = useRef<HTMLInputElement>(null);
+  const passwordRef = useRef<HTMLInputElement>(null);
+  const confirmPasswordRef = useRef<HTMLInputElement>(null);
+  const genderRef = useRef<HTMLSelectElement>(null);
+  const termsRef = useRef<HTMLInputElement>(null);
+  const imageRef = useRef<HTMLInputElement>(null);
+  const countryRef = useRef<HTMLInputElement>(null);
+
+  const [errors, setErrors] = useState<{ [key: string]: string }>({});
```

Comment:
it's better to make types for errors like type FormErrors = Partial<Record<
  'name' | 'age' | 'email' | 'confirmPassword' ...etc,
  string
>>;